src/components/interactive-image/interactive-image.tsx

```diff
@@ -0,0 +1,222 @@
+
+@Component({
+  shadow: false,
+  styleUrl: "interactive-image.scss",
+  tag: "gx-interactive-image"
+})
+export class InteractiveImage implements GxComponent {
+  @Element() element: HTMLGxInteractiveImageElement;
+
+  /**
+   * If this property is true, the user can copy the image to the clipboard (Only for iOS, see image below).
+   */
+  @Prop() enableCopyToClipboard: false;
+
+  /**
+   * True/False. If this property is true, the user can zoom in/out on the image.
+   */
+  @Prop() enableZoom = false;
+
+  /**
+   * Indicates how much you can enlarge an image. (Percentage) _Note: 100% = Normal size_.
+   */
+  @Prop() zoom = 100;
+
+  /**
+   * Indicates how much you can enlarge an image with reference to the original size of it or the size of the controller. (Percentage)
+   */
+  @Prop() maxZoomRelativeTo: number;
+
+  /**
+   * Lets you specify the "src" of the img.
+   */
+  @Prop() readonly src = "";
+
+  @State() mouseOver = false;
+
+  @State() zoomedPositionX: number;
+
+  @State() zoomedPositionY: number;
+
+  @Watch("mouseOver")
+  mouseOverHandler() {
+    this.editingOverClass();
+  }
+
+  private addEvent(
+    element: HTMLElement,
+    eventToListen: string,
+    callbackFunction,
+    passive: boolean
+  ) {
+    element.addEventListener(eventToListen, callbackFunction),
+      passive && { passive: true };
+  }
+
+  private calculateZoomedPosition(overPosition: number, elementSize: number) {
+    const SCALE = this.zoom / 100;
+    const HALF_SIZE_PERCENTAGE = 50;
+    const HALF_SIZE_PIXELS = (elementSize * HALF_SIZE_PERCENTAGE) / 100;
+    return -(overPosition - HALF_SIZE_PIXELS) * (SCALE - 1);
+  }
+
+  private calculateZoomTouch(preCalculateValue: number, zoom: number) {
+    const SCALE = zoom / 100;
+    return preCalculateValue / SCALE;
+  }
+
+  private checkZoomFeature() {
+    const zooming = this.zoomFeature;
+    if (this.enableZoom) {
+      const img = this.element.querySelector("img");
+      this.addEvent(
+        img,
+        zooming.over.withMouse,
+        zooming.over.mouseBehaivor,
+        true
+      );
+      this.addEvent(
+        img,
+        zooming.over.withTouch,
+        zooming.over.touchBehaivor,
+        true
+      );
+      this.addEvent(img, zooming.out.withMouse, zooming.out.behaivor, true);
+      this.addEvent(img, zooming.out.withTouch, zooming.out.behaivor, true);
+    } else {
+      const img = this.element.querySelector("img");
+      this.removeEvent(img, zooming.over.withMouse, zooming.over.mouseBehaivor);
+      this.removeEvent(img, zooming.over.withTouch, zooming.over.touchBehaivor);
+      this.removeEvent(img, zooming.out.withMouse, zooming.out.behaivor);
+      this.removeEvent(img, zooming.out.withTouch, zooming.out.behaivor);
+    }
+  }
+
+  private correctZoomValue() {
+    if (this.zoom < 100) {
+      console.warn("Zoom value cannot be lower than 100");
+      this.zoom = 100;
+    }
+  }
+
+  private editingOverClass() {
```

Comment:
Nitpicking here, but I feel we can come up with a better name for this method.

Reply:
That method is useless now. It was removed in d1cafe5